cbopensource/connectors/taxii/cybox_parse.py

```diff
@@ -177,10 +207,33 @@ def cybox_parse_observable(observable, indicator, timestamp, score):
                                 'timestamp': timestamp,
                                 'link': link,
                                 'score': score})
+        if props.category == 'ipv6-addr' and props.address_value:
+            iocs = {'ipv6': []}
+
+            #
+            # Sometimes props.address_value.value is a list vs a string
+            #
+            if type(props.address_value.value) is list:
```

Comment:
This function is going beyond it's responsibility and is no longer clear.  This portion of code is a good candidate to be refactored into its own function.